File: Data Structures & Algorithms/subsets/submission-4.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>>ans;
        vector<int>temp;
        for(int i=0;i<(1<<nums.size());++i){
            temp.clear();
            for(int j=0;j<nums.size();++j){
                if(i & (1<<j)) temp.emplace_back(nums[j]);
            }
            ans.emplace_back(temp);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/permutations/submission-4.py
import copy
class Solution:
    ans = []
    def generate(self,nums, temp_nums, vis):
        if len(temp_nums)==len(nums):
            self.ans.append(copy.copy(temp_nums))
        for i in range(len(nums)):
            if vis[i]==False:
                vis[i] = True
                temp_nums.append(nums[i])
                self.generate(nums,temp_nums,vis)
                temp_nums.pop()
                vis[i]=False
        

    def permute(self, nums: List[int]) -> List[List[int]]:
        self.ans = []
        self.generate(nums,[],[False for _ in range(len(nums))])
        return self.ans